CampaignAutomation/Tests/CampaignAutomationImplementingTests.swift

```diff
@@ -82,33 +82,66 @@ class CampaignAutomationImplementingTests: XCTestCase {
             source: .server,
             dependentKeys: nil)
         graphQLManager.resultReturns = .success(result)
-
-        let expectation = XCTestExpectation(description: "Completion fulfilled")
         sut.generateCreatables(for: "", factories: [.banner: MockBannerFactory()]) { result in
             guard case let .success(creatables) = result, let banner = creatables.first as? MockBannerCreatable else {
                 return XCTFail("Test failed")
             }
             XCTAssertEqual(creatables.count, 1)
             XCTAssertEqual(banner.title, "Banner title")
-            expectation.fulfill()
         }
-        wait(for: [expectation], timeout: 0.01)
     }
 
     func test_generateCreatables_failure() {
         graphQLManager.resultReturns = .failure(DummyError.failure)
-
-        let expectation = XCTestExpectation(description: "Completion fulfilled")
         sut.generateCreatables(for: "", factories: [.banner: MockBannerFactory()]) { result in
             guard case let .failure(error) = result else {
                 return XCTFail("Test failed")
             }
             XCTAssertEqual((error as? DummyError), DummyError.failure)
             // ensure that cache was accessed after server query failure
             XCTAssertEqual(self.graphQLManager.numberOfQueriesCalled, 2)
-            expectation.fulfill()
         }
-        wait(for: [expectation], timeout: 0.01)
+    }
+
+    func test_fechData_success() {
+        let items = responseItems.map {
+            QueryDataType.GetContentByExternalId.Item(
+                contentType: ApolloTypeCA.ContentType(rawValue: $0.contentType ?? ""),
+                data: $0.data)
+        }
+        let getContentByExternalId = QueryDataType.GetContentByExternalId(campaignId: "1",
+                                                                          items: items)
+        let data = QueryDataType(getContentByExternalId: getContentByExternalId)
+        let result = GraphQLResult<QueryDataType>(
+            data: data,
+            extensions: nil,
+            errors: nil,
+            source: .server,
+            dependentKeys: nil)
+        graphQLManager.resultReturns = .success(result)
+        sut.fetchData(for: "") { result in
```

Comment:
This test will pass if the callback is not called (false positive test). You can save `result` in a variable and make all assertions after the `fetchData` method.

Reply:
Good catch! I've updated it, thank you!